commit 2c9450dceac031d05daa8b5dc292a1b1f775c767
Author: Dom Cobley <dc4@broadcom.com>
Date:   Mon Apr 30 14:45:52 2012 +0100

    added support for TT in the USB driver. Thanks Naren

Index: linux-3.2.46/drivers/usb/gadget/Kconfig
===================================================================
--- linux-3.2.46.orig/drivers/usb/gadget/Kconfig	2013-07-26 19:32:11.000000000 +0000
+++ linux-3.2.46/drivers/usb/gadget/Kconfig	2013-07-26 19:32:51.000000000 +0000
@@ -552,14 +552,6 @@
 	depends on USB_GADGET
 	depends on USB_GADGET_DUALSPEED
 
-config USB_GADGET_SNPS_DWC_OTG
-	boolean "Synopsys Driver for DWC_otg Controller"
-	depends on USB && EXPERIMENTAL
-	select USB_OTG
-	select USB_GADGET_DUALSPEED
-	help
-	   Selects the Synopsys Driver for the DWC_otg Controller.
-
 config USB_DWC_OTG_LPM
 	boolean "Enable LPM support"
 	depends on USB && EXPERIMENTAL
Index: linux-3.2.46/drivers/usb/host/dwc_otg/dwc_otg_hcd_linux.c
===================================================================
--- linux-3.2.46.orig/drivers/usb/host/dwc_otg/dwc_otg_hcd_linux.c	2013-07-26 19:32:45.000000000 +0000
+++ linux-3.2.46/drivers/usb/host/dwc_otg/dwc_otg_hcd_linux.c	2013-07-26 19:32:51.000000000 +0000
@@ -417,6 +417,9 @@
 
 	hcd->regs = otg_dev->base;
 
+	/* Integrate TT in root hub */
+	hcd->has_tt = 1;
+
 	/* Initialize the DWC OTG HCD. */
 	dwc_otg_hcd = dwc_otg_hcd_alloc_hcd();
 	if (!dwc_otg_hcd) {
